docs/reference/syntax-guide.rst

```diff
@@ -4,6 +4,72 @@ Syntax Guide
 
 Examples are mostly adapted from the Idris tutorial.
 
+Source File Structure
+---------------------
+
+Source files consist of:
+
+1. An optional :ref:`syntax-module-headers`.
+2. Zero or more :ref:`syntax-imports`.
+3. The file's content, which is a mix of :ref:`syntax-variables`,
+   :ref:`syntax-data-types`, :ref:`syntax-directives`, etc.
+
+.. Should that last bullet become more specific?  Terminologically I'm not even
+   certain what the top-level entities in an Idris file are properly called.
+
+For example:
+
+.. code:: idris
+
+    module MyModule   -- module header
+
+    import Data.Vect  -- an import
+
+    %default total    -- a directive
+
+    foo : Nat         -- a declaration
+    foo = 5
+
+.. _syntax-module-headers:
+
+Module Header
+~~~~~~~~~~~~~
+
+The module header appears at the top of the file.  Each file can contain only
+one module, but content within the file may be broken into multiple namespaces.
+
+Unlike most Idris declaration headers, the module header does not end in the
```

Comment:
Shouldn't it mention the `module` keyword here? I'd phrase it

A module header consists of the keyword `module` followed by the name of the module. Hierarchical module names are specified with `.` as a separator. Unlike some other languages, Idris modules headers are not terminated with the keyword `where`.